packages/components/src/Combobox/components/ComboboxContent/ComboboxHeader.tsx

```diff
@@ -16,17 +19,25 @@ export function ComboboxHeader(props: ComboboxHeaderProps): JSX.Element {
   const label = getLabel(hasSelected, props.selectedCount, props.subjectNoun);
   const handleSelectAll = hasSelected ? props.onClearAll : props.onSelectAll;
 
+  const searchTermMatches = props.options.some(option =>
+    option.label.toLowerCase().includes(props.searchValue.toLowerCase()),
+  );
+
+  const showClearButton = hasSelected || searchTermMatches;
```

Comment:
a lil nitpick here 💅 

this isn't always going to be a `ClearButton`, so a more flexible name could help communicate that

perhaps..... `showAction`? `showButton`? something like that, that's more agnostic of the function/behavior since that is dynamic

Reply:
`showAction` - thank you! Did that here: fda87b77dc2f1061717036b0909a17b0bf0c9c21